### PR TITLE
Refactor, test and update app

### PR DESCRIPTION
**Pull request by bettblake08**
**What does this PR do?**
This PR introduces changes to the codebase that include refactored code, update of search filter input component, update of article component and article page as well as the token refresh process.

**Description of tasks to be completed?**
- remove window href uses
- add search filter status
- disable search bar when on discover page
- remove reducer suffix from redux initial state
- remove __test__ folders
- refactor scss files in articles folder
- add refresh token process for authentication required api call
- add image and body css for article page
- add active tag highlighting on discover page
- fix discover page to filter articles co-operatively
- add caret to search bar type dropdown

**How should this be manually tested?**
- Clone the repo and cd into the project root folder
- Switch to the relevant branch `git checkout ch-implement-feedback`
- Run `npm install` to update your packages.
- Start the server with `npm start`
- In your favorite browser, navigate to `http://localhost:3000`

### Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings

### CODE REVIEW

File: src/actions/bookmarkArticleActions.js [hunk 1]
```diff
@@ -22,24 +22,38 @@ export const unBookmarkArticleFailure = data => ({
...
+  slug,
+  onSuccess = () => { },
+  onFailure = () => { }
+) => (dispatch) => {
   bookmarkArticleApi.unBookmarkApi(slug).then((response) => {
     if (response.success) {
       dispatch(unBookmarkArticleSuccess(response.content));
-      dispatch(getSingleArticle(slug));
+      // dispatch(getSingleArticle(slug));
+      onSuccess();
```
Comment (denomash) [inline src/actions/bookmarkArticleActions.js:62]: @bettblake08 Kindly remove this commented code.

File: src/actions/bookmarkArticleActions.js [hunk 2]
```diff
@@ -22,24 +22,38 @@ export const unBookmarkArticleFailure = data => ({
...
     } else {
-      dispatch(unBookmarkArticleFailure(response.error.message));
+      checkIfUnauthorized(response, dispatch, () => unBookmarkArticle(slug));
+      onFailure();
+      // dispatch(unBookmarkArticleFailure(response.error.message));
     }
```
Comment (denomash) [inline src/actions/bookmarkArticleActions.js:66]: @bettblake08 Kindly remove this commented code too.

**Review by denomash — CHANGES_REQUESTED** [review]

**Review by Darius-Ndubi — APPROVED** [review]
👍

**Review by denomash — APPROVED** [review]
👍

**Review by davishooly — APPROVED** [review]
👍

**Pull request merged**